src/openforms/forms/tests/test_migrations.py

```diff
@@ -287,3 +287,165 @@ def test_migration_fixes_issues(self):
+                "label": "Text Area 1",
+                "conditional": {"eq": "0", "show": True, "when": "number"},
+                "clearOnHide": True,
+            },
+            {
+                "key": "textArea2",
+                "type": "textarea",
+                "label": "Text Area 2",
+                "conditional": {"eq": "0.555", "show": True, "when": "number"},
+                "clearOnHide": True,
+            },
+            {
+                "key": "textArea3",
+                "type": "textarea",
+                "label": "Text Area 3",
+                "conditional": {"eq": "", "show": None, "when": ""},
+                "clearOnHide": True,
+            },
+            {
+                "key": "textArea4",
+                "type": "textarea",
+                "label": "Text Area 4",
+                "conditional": {"when": "currency", "eq": "0.55", "show": True},
+                "clearOnHide": True,
+            },
+            {
+                "key": "textArea5",
+                "type": "textarea",
+                "label": "Text Area 5",
+                "conditional": {"eq": "1.00", "when": "currency", "show": True},
+                "clearOnHide": True,
+            },
+            {
+                "key": "textArea6",
+                "type": "textarea",
+                "label": "Text Area 6",
+                "conditional": {"eq": "1.00", "when": "textfield", "show": True},
+                "clearOnHide": True,
+            },
+            {
+                "key": "repeatingGroup",
+                "type": "editgrid",
+                "label": "Repeating group",
+                "components": [
+                    {
+                        "key": "textArea7",
+                        "type": "textarea",
+                        "label": "Text Area 7",
+                        "conditional": {"eq": "0", "show": True, "when": "number"},
+                        "clearOnHide": True,
+                    },
+                ],
+            },
+        ]
+        FormDefinition.objects.create(
+            name="broken", configuration={"components": bad_config}
+        )
+
+    def test_conditionals_are_fixed(self):
+        FormDefinition = self.apps.get_model("forms", "FormDefinition")
+        fixed_components = FormDefinition.objects.get().configuration["components"]
+
+        self.assertTrue(isinstance(fixed_components[3]["conditional"]["eq"], int))
+        self.assertTrue(isinstance(fixed_components[4]["conditional"]["eq"], float))
+        self.assertTrue(isinstance(fixed_components[5]["conditional"]["eq"], str))
+        self.assertTrue(isinstance(fixed_components[6]["conditional"]["eq"], float))
+        self.assertTrue(isinstance(fixed_components[7]["conditional"]["eq"], float))
+        self.assertTrue(isinstance(fixed_components[8]["conditional"]["eq"], str))
+        self.assertTrue(
+            isinstance(fixed_components[9]["components"][0]["conditional"]["eq"], int)
+        )
```

Comment:
can't we compare with the exact value rather than isinstance checks?

What if a number accidentally gets misinterpreted as hex or octals (wild guessing here on my part)?

Reply:
I wanted to avoid doing equalities with floats, and then I kept the other consistent. But I can compare with the actual value.